check50/__main__.py

```diff
@@ -203,7 +206,55 @@ def raise_invalid_slug(slug, offline=False):
         msg += _("\nIf you are confident the slug is correct and you have an internet connection," \
                 " try running without --offline.")
 
-    raise internal.Error(msg)
+    raise _exceptions.Error(msg)
+
+
+def process_args(args):
+    """Validate arguments and apply defaults that are dependent on other arguments"""
+
+    # dev implies offline, verbose, and log level "INFO" if not overwritten
+    if args.dev:
+        args.offline = True
+        args.verbose = True
+        if "ansi" in args.output:
+            args.ansi_log = True
+
+    # offline implies local
+    if args.offline:
+        args.no_install_dependencies = True
+        args.no_download_checks = True
+        args.local = True
+
+    if not args.log_level:
+        args.log_level = LogLevel.WARNING
+
+    # Setup logging for lib50
+    setup_logging(args.log_level)
+
+    # Warning in case of running remotely with no_download_checks or no_install_dependencies set
+    if not args.local:
+        useless_args = []
+        if args.no_download_checks:
+            useless_args.append("--no-downloads-checks")
+        if args.no_install_dependencies:
+            useless_args.append("--no-install-dependencies")
+
+        if useless_args:
+            LOGGER.warning(_("You should always use --local when using: {}").format(", ".join(useless_args)))
+
+    # Filter out any duplicates from args.output
+    seen_output = []
+    for output in args.output:
+        if output in seen_output:
+            LOGGER.warning(_("Duplicate output format specified: {}").format(output))
+        else:
+            seen_output.append(output)
+
+    args.output = seen_output
+
+    if args.ansi_log and "ansi" not in seen_output:
+        LOGGER.warning(_("--ansi-log has no effect when ansi is not among the output formats"))
```

Comment:
I honestly still don't like these useless arguments and would much rather exit with an error code if an argument that's useless or shouldn't be used is passed.

Reply:
I disagree, I think a warning is in place here because these command-line options are harmless. Misuse (useless use) of them does not change your environment, submit code that you wouldn't want to submit or anything of the sorts.

My thinking is, we should only switch to errors in case misuse of a command line option can have an unintended consequence.

Reply:
It just seems to me that it's pretty much always better to enforce the rules. When an argument is either misused, may have unintended behaviors, or just doesn't make sense in a certain context it just seems better exit with an error code and force the user to remove it because there's absolutely no reason to let the user pass it when it's not useful and not needed. Just like that Docker CLI example that you brought up and it turned out they were enforcing it. Can we please change this behavior? :sob: